docs/topics/control-flow.md

```diff
@@ -50,71 +50,152 @@ val max = if (a > b) {
+
+`when` matches its argument against all branches sequentially until some branch condition is satisfied.
+
+You can use `when` in a few different ways. Firstly, you can use `when` either as an **expression** or as a **statement**.
+As an expression, `when` returns a value that you can use later in your code. As a statement, `when` completes an action
+without returning anything of further use:
+
+<table>
+   <tr>
+       <td>Expression</td>
+       <td>Statement</td>
+   </tr>
+   <tr>
+<td>
+
+```kotlin
+// Returns a string assigned to the 
+// text variable
+val text = when (x) {
+    1 -> "x == 1"
+    2 -> "x == 2"
+    else -> "x is neither 1 nor 2"
+}
+```
+
+</td>
+<td>
+
+```kotlin
+// Returns nothing but triggers a 
+// print statement
 when (x) {
     1 -> print("x == 1")
     2 -> print("x == 2")
-    else -> {
-        print("x is neither 1 nor 2")
-    }
+    else -> print("x is neither 1 nor 2")
 }
 ```
 
-`when` matches its argument against all branches sequentially until some branch condition is satisfied.
+</td>
+</tr>
+</table>
 
-`when` can be used either as an expression or as a statement. If it is used as an expression, the value
-of the first matching branch becomes the value of the overall expression. If it is used as a statement, the values of
-individual branches are ignored. Just like with `if`, each branch can be a block, and its value
-is the value of the last expression in the block.
+Secondly, you can use `when` with or without a subject.
 
-The `else` branch is evaluated if none of the other branch conditions are satisfied.
+<table>
+   <tr>
+       <td>With subject <code>x</code></td>
+       <td>Without subject</td>
+   </tr>
+   <tr>
+<td>
 
-If `when` is used as an _expression_, the `else` branch is mandatory,
-unless the compiler can prove that all possible cases are covered with branch conditions,
-for example, with [`enum` class](enum-classes.md) entries and [`sealed` class](sealed-classes.md) subtypes).
+```kotlin
+when(x) { ... }
+```
+
+</td>
+<td>
+
+```kotlin
+when { ... }
+```
+
+</td>
+</tr>
+</table>
+
+> Where possible, we recommend using `when` with a subject. Using `when` with a subject makes your code easier to read
+> and maintain by clearly showing what is being checked.
+> 
+{style="note"}
+
+Depending on how you use `when`, there are different requirements for whether you need to cover all possible cases in your
+branches.
+
+If you use `when` as a statement, you don't have to cover all possible cases:
+
+```kotlin
+fun main() {
+    //sampleStart
+    val x = 2
+    when (x) {
+        // Not all cases are covered
+        1 -> print("x == 1")
+        2 -> print("x == 2")
+    }
+    //sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-when-statement"}
+
+In a `when` statement, the values of individual branches are ignored. Just like with `if`, each branch can be a block, 
```

Comment:
Can we add information about what happens when not all the cases are covered and the value doesn't match any of the branches? It is not an error, but is there something returned? 

For example:

fun main() {
    //sampleStart
    val x = 3
    when (x) {
        // Not all cases are covered
        1 -> print("x == 1")
        2 -> print("x == 2")
    }
    //sampleEnd
}

Reply:
I updated the text to try and explain this better. What do you think? :)